docs/stacks-userguide.md

```diff
@@ -0,0 +1,807 @@
+```
+myservice1:
+    image: appcelerator/pinger:latest
+    environment:
+      myvar1:  myvalue1
+      myvar2: myvalue2
+```
+
+Sample with list
+```
+myservice1:
+    Image: appcelerator/pinger:latest
+    environment:
+      - “myvar1= myvalue1”
+      - “myvar2=myvalue2”
+```
+
+### 2.4.4 labels, service item 
+
+
+Adds labels to the service (visible using docker service inspect)
+By default no labels are added
+
+
+Two syntaxes are supported map or list:
+
+
+Sample with map
+```
+myservice1:
+    image: appcelerator/pinger:latest
+    labels:
+      mylabel1:  myvalue1
+      mylabel2: myvalue2
+```
+
+Sample with list
+```
+myservice1:
+    image: appcelerator/pinger:latest
+    labels:
+      - “mylabel1= myvalue1”
+      - “mylabel2=myvalue2”
+```
+
+### 2.4.5 containers labels, service item
+
+
+Adds labels to all the containers created for the service (visible using docker inspect)
+By default no container labels are added
+
+
+Two syntaxes are supported map or list:
+
+
+Sample with map
+```
+myservice1:
+    image: appcelerator/pinger:latest
+    container_labels:
+      mylabel1:  myvalue1
+      mylabel2: myvalue2
+```
+
+Sample with list
+```
+myservice1:
+    image: appcelerator/pinger:latest
+    container_labels:
+      - “mylabel1= myvalue1”
+      - “mylabel2=myvalue2”
+```
+
+### 2.4.6 public, service item
+
+
+Define how the service can be requested from outside the stack.
+There are two ways:
+- Using name
+- Using publish port
+
+For each, a protocol can be defined, tcp by default
+
+
+
+
+##### Using name:
+
+
+Several public names can be set for a service. Each of them defines a relation between an internal service port and a logical name that the stack HAProxy can use to request the service (see HAProxy chapter for more details)
+
+
+Sample:
+```
+myservice1:
+    image: appcelerator/pinger:latest
+    public:
+      - name: myexternalname
+        protocol: tcp
+        internal_port: 3000
+```
+
+Then: `http://myexternalname.[domain]/xxx`
+will be load-balanced to: `[myService containers addresses]:3000/xxx`
+
+
+
+
+##### Using publish_port:
+
+
+Several publish ports can be set for a service. Each of them defined a relation between the publish port and an internal service port. On all swarm node any http request on the publish port will be rerouted to the service internal port (see Network chapter for more details)
+
+
+Sample:
+```
+myservice1:
+    image: appcelerator/pinger:latest
+    public:
+       - publish_port: 3001
+         internal_port: 3000
+```
+
+Then: `http://[any swarm node address]:3001/xxx`
+will be load-balanced to: `[myService container addresses]:3000/xxx`
+
+
+##### Using both name and publish_port
+
+
+It’s possible to use name and publish port on the same internal port. They should use the same protocol.
+
+
+Sample:
+```
+myservice1:
+    image: appcelerator/pinger:latest
+    public:
+      -name: myexternalname
+       publish_port: 3001
+       internal_port: 3000
+```
```

Comment:
curious why we're using `publish_port` and `internal_port`? why not use a more docker standard syntax `internal_port:external_port` such as:

```yaml
myservice1:
   image: appcelerator/pinger:latest
   public:
       - name: myexternalname
         port: 3000:3001
```